### ISSUE TITLE
Writing log file for analyze.sh

### ISSUE THREAD

**Issue by dbetchkal** [NINAnor/snowmobile_analyzer#1]
Greetings, I want to confirm that I have been able to follow the README instructions for Windows+Docker to produce the expected result:
`.snowmobile_analyzer/example/SNOWMOBILE_RESULTS/example_audio_ANALYZED.csv`

However, when trying to test any other audio the bash shell script fails. Unfortunately the traceback can be seen for only a very short time (<1 second). I am wondering if it would be possible to write out print statements and the traceback from the console to a log file? That way it would be easier to debug when `cmd` crashes. I can work on a solution independently, but no doubt you will be faster @BenCretois.

**Comment by BenCretois** [comment]
Fixed with commit [2d6f577](https://github.com/NINAnor/snowmobile_analyzer/commit/2d6f57738b1c2b20ce1d0e291f4ed049a7464917)

Strange that it does not work for other files, it works fine on my system. The repository now contains a `logs` folder  in which a `logfile.log` is created.

**Comment by dbetchkal** [comment]
Greetings @BenCretois unfortunately this fix (though speedy) does not resolve the issue for me. No log file is written. After some painstaking efforts with the printscreen button I have been able to capture the error message:
![image](https://github.com/NINAnor/snowmobile_analyzer/assets/17891020/c43ea8d3-a8bc-4f0a-923c-7555bfa8160f)

After a bit of searching I believe the issue may be related to the formatting of the `-v` flag paths:
```
    -v ./logs:/app/logs \
    -v $FOLDER_TO_EXPOSE:/data \
    snowmobile \
```
Apparently they aren't formatted correctly for Windows. I am new enough to `docker run` that I don't quite grasp the syntax of the flags. Any help here would be appreciated.

**Comment by BenCretois** [comment]
Hi @dbetchkal ! Thanks for reporting the issue :)

So the `-v` flag is made to expose a volume. The docker container itself can't access your local folders so they need to be exposed using `-v`.  

I am unable to test the application on Windows as we're just using Ubuntu. However on [StackOverflow it seems that someone got the same problem](https://stackoverflow.com/questions/63213598/docker-network-drive-extra-string-is-not-a-valid-windows-path).

Moreoever, the `analyze.sh` is a bash script, do you have a bash interpreter on your Windows?

If anything you could also try to use `docker run` directly instead of relying on the `analyze.sh` script:

```
docker run \
    --rm \
    --gpus all \
    -v ./logs:/app/logs \
    -v YOUR/PATH/TO/AUDIO:/data \
    ghcr.io/ninanor/snowmobile_analyzer:main \
    --input /data/YOUR_FILE_NAME
```

Let me know if you manage to run the application!

**Comment by dbetchkal** [comment]
Hello again, thank you for the advice. I can confirm that `docker run` from the command line appears to advance this process further. Now there appears to be a missing file? `snowmobile_model.pth`?

Here is the full traceback:
```
C:\Users\DBetchkal\PythonScripts\3_GITHUB_REPOSITORIES\snowmobile_analyzer>docker run --rm --gpus all -v ./logs:/app/logs -v C://Users//DBetchkal//Desktop//NPS_tests:/data ghcr.io/ninanor/snowmobile_analyzer:main --input C://Users//DBetchkal//Desktop//NPS_tests//DENADUNK_20120318_050147.mp3
Skipping virtualenv creation, as specified in config file.
Traceback (most recent call last):
  File "src/predict.py", line 198, in <module>
    model = initModel(model_path=model_path, device=device)
  File "src/predict.py", line 20, in initModel
    model = torch.load(model_path)
  File "/usr/local/lib/python3.8/site-packages/torch/serialization.py", line 699, in load
    with _open_file_like(f, 'rb') as opened_file:
  File "/usr/local/lib/python3.8/site-packages/torch/serialization.py", line 231, in _open_file_like
    return _open_file(name_or_buffer, mode)
  File "/usr/local/lib/python3.8/site-packages/torch/serialization.py", line 212, in __init__
    super(_open_file, self).__init__(open(name, mode))
FileNotFoundError: [Errno 2] No such file or directory: '/app/audioclip/assets/snowmobile_model.pth'
```

**Comment by BenCretois** [comment]
Hi Davyd, I reopened the issue since it's not solved :)

The error you get is totally my fault, the path to the `snowmobile_model.pth` is not set properly in the script. The path should be `/app/assets/snowmobile_model.pth` instead of `/app/audioclip/assets/snowmobile_model.pth`.

This should be fixed by [commit e29a450](https://github.com/NINAnor/snowmobile_analyzer/commit/e29a45017596965d51ab1d6c86443b4f82215e6f). Let me know if the fix solves your problems!

Note that you will have to pull the Docker image once again.

**Comment by dbetchkal** [comment]
OK, after pulling the Docker image and re-running the same `docker run` command, I now appear to be [missing a *.txt.gz* file related to `openai`](https://github.com/openai/CLIP/blob/main/clip/bpe_simple_vocab_16e6.txt.gz)? Here is the traceback:

```C:\Users\DBetchkal\PythonScripts\3_GITHUB_REPOSITORIES\snowmobile_analyzer> docker run --rm --gpus all -v ./logs:/app/logs -v C://Users//DBetchkal//Desktop//NPS_tests:/data ghcr.io/ninanor/snowmobile_analyzer:main --input C://Users//DBetchkal//Desktop//NPS_tests//DENADUNK_20120318_050147.mp3
Skipping virtualenv creation, as specified in config file.
Traceback (most recent call last):
  File "src/predict.py", line 198, in <module>
    model = initModel(model_path=model_path, device=device)
  File "src/predict.py", line 20, in initModel
    model = torch.load(model_path)
  File "/usr/local/lib/python3.8/site-packages/torch/serialization.py", line 712, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/usr/local/lib/python3.8/site-packages/torch/serialization.py", line 1046, in _load
    result = unpickler.load()
  File "/usr/local/lib/python3.8/site-packages/torch/serialization.py", line 1039, in find_class
    return super().find_class(mod_name, name)
  File "/app/audioclip/model/__init__.py", line 3, in <module>
    from .audioclip import AudioCLIP
  File "/app/audioclip/model/audioclip.py", line 7, in <module>
    from model.clip.clip import tokenize
  File "/app/audioclip/model/clip/clip.py", line 18, in <module>
    _tokenizer = _Tokenizer()
  File "/app/audioclip/utils/simple_tokenizer.py", line 68, in __init__
    merges = gzip.open(bpe_path).read().decode("utf-8").split('\n')
  File "/usr/local/lib/python3.8/gzip.py", line 58, in open
    binary_file = GzipFile(filename, gz_mode, compresslevel)
  File "/usr/local/lib/python3.8/gzip.py", line 173, in __init__
    fileobj = self.myfileobj = builtins.open(filename, mode or 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/app/audioclip/utils/../assets/bpe_simple_vocab_16e6.txt.gz'```

**Comment by BenCretois** [comment]
Hi Davyd

Okay, so I modified also the path for the `.gz` path. I also double check if there were other paths that could prove problematic but haven't found any others so far.  

The problem should -hopefully- be fixed with [dd6ffed](https://github.com/NINAnor/snowmobile_analyzer/commit/dd6ffeda44f25de1990ca522197b3cd77d7a21df)

Let me know and sorry for the trouble!
